client/lxd_instances.go

```diff
@@ -1134,6 +1135,40 @@ func (r *ProtocolLXD) ExecInstance(instanceName string, exec api.InstanceExecPos
 			}
 		}
 
+		if exec.RecordOutput && (args.Stdout != nil || args.Stderr != nil) {
+			err = op.Wait()
+			if err != nil {
+				return nil, err
+			}
+
+			opAPI = op.Get()
+			outputFiles := map[string]string{}
+			outputs, ok := opAPI.Metadata["output"].(map[string]any)
+			if ok {
+				for k, v := range outputs {
+					outputFiles[k] = v.(string)
+				}
+			}
+
+			if outputFiles["1"] != "" {
+				reader, _ := r.GetInstanceExecOutputLogFile(instanceName, filepath.Base(outputFiles["1"]))
+				if args.Stdout != nil {
+					_, _ = io.Copy(args.Stdout, reader)
+				}
+
+				_ = r.DeleteInstanceExecOutputLogFile(instanceName, filepath.Base(outputFiles["1"]))
+			}
+
+			if outputFiles["2"] != "" {
+				reader, _ := r.GetInstanceExecOutputLogFile(instanceName, filepath.Base(outputFiles["2"]))
+				if args.Stderr != nil {
+					_, _ = io.Copy(args.Stderr, reader)
+				}
+
+				_ = r.DeleteInstanceExecOutputLogFile(instanceName, filepath.Base(outputFiles["2"]))
```

Comment:
Thanks Mark

Reply:
@tomponline for the [errcheck linter](https://golangci-lint.run/usage/linters/#errcheck) should we enable `check-blank`?

This would mean that any explicitly ignored errors would need to be accompanied by a `//nolint:errcheck` directive (which would also be a convenient place to explain why the error is ignored).

Reply:
Perhaps. Was there a reason we didn't use that originally vs the approach we did go for (which I believe had custom rules)?

Reply:
No I think we just went with the defaults in most cases as there is a lot lot lot of configuration for all these linters. I think this one makes sense though.

Reply:
Dont we have a missing error handler check already though?

Reply:
Yes but by default it skips errors that are explicitly ignored with underscores. E.g.
```golang
package main

import (
    "errors"
)

func returnsAnError() error {
    return errors.New("error")
}

func main() {
    returnsAnError() // is caught by errcheck by default
    _ = returnsAnError() // is allowed by default    
}
```

So setting `check-blank` to true will mean all errors that are not checked will need an `//nolint:errcheck` regardless of whether they are explicitly ignored.

Reply:
Oh right ofcourse. Sorry. I was at cross purposes. 

How many of the explicit definitions do we have now? Wondering how big a change this will be?

Reply:
I think we'll pass on that then for now. The requiring of explicit ignoring errors `_` character has done its job here of drawing your attention so its fixed. 

That there are that many of them suggests that this isn't an exception, but quite a common pattern/need. So having a linter require explaining that reasoning over and over doesn't seem that useful to me.

What do you think?

Reply:
I'm honestly not sure. On scanning a few of these it seems we have a few different cases:
1. Errors that are ignored for a clear reason e.g. `regexp.MatchString` never returns an error.
2. Ignored errors but the reason isn't clear e.g. https://github.com/canonical/lxd/blob/afc1356c2762b16f87f47a13ff714d7bfe54210e/lxd/storage/filesystem/fs.go#L132
3. Ignored errors that could be mistakes e.g. https://github.com/canonical/lxd/blob/main/lxd/storage/drivers/driver_btrfs.go#L361
4. Deferred functions that must be wrapped in order to explicitly ignore an error e..g. `defer func() { _ = rows.Close() }()`. These could be `defer rows.Close() //nolint:errcheck`.

In general I think we should be at least logging errors when they occur, even if reverting changes, and if we are ignoring an error then having to add the //nolint:errcheck would help explain to the reader what the reason is.

Additionally it looks like we can specify a list of functions to exclude from checking (see https://golangci-lint.run/usage/linters/#errcheck).

So maybe this is something we can revisit do when we have a little more time on our hands :laughing: